src/modules/esl-media/providers/vimeo-provider.ts

```diff
@@ -0,0 +1,191 @@
+import axios from 'axios';
+
+import {BaseProvider, PlayerStates} from '../core/esl-media-provider';
+import {randUID} from '../../esl-utils/misc/uid';
+import {loadScript} from '../../esl-utils/dom/script';
+import {memoize, tryUntil} from '../../esl-utils/all';
+
+import type {Player, Options, Error} from  '@vimeo/player';
+import type {MediaProviderConfig, ProviderObservedParams} from '../core/esl-media-provider';
+import type {ESLMedia} from '../core/esl-media';
+
+
+@BaseProvider.register
+export class VimeoProvider extends BaseProvider {
+  constructor(component: ESLMedia, config: MediaProviderConfig){
+    super(component, config);
+    this.config = config;
+    this.component = component;
+  }
+  static readonly providerName: string = 'vimeo';
+
+  static readonly providerRegexp = /(http|https)?:\/\/(www\.|player\.)?vimeo\.com\/(?:channels\/(?:\w+\/)?|groups\/([^/]*)\/videos\/|video\/|)(\d+)(?:|\/\?)/i;
+
+  protected _el: HTMLDivElement | HTMLIFrameElement;
+  protected _api: Player;
+  protected _state: PlayerStates;
+  protected _duration: number;
+  protected _lastCurrentTime: number = 0;
+  protected _defaultAspectRatio: number;
+  protected _startData: number = 0;
+
+
+  // static async parseUrl(url: string) {
+  //   const match = url.match(this.providerRegexp);
+  //   if (!match) return null;
+  //   const mediaId = [].pop.call(match);
+  //   const {data} = await axios.get(`https://vimeo.com/api/oembed.json?url=${url}`);
+  //   url = 'https://player.vimeo.com/video/' + mediaId;
+  //   console.log(data);
+  //   return isNaN(mediaId) ? null : {mediaId};
+  // }
+
+  static parseUrl(url: string) {
+    const match = url.match(this.providerRegexp);
+    if (!match) return null;
+    const mediaId = [].pop.call(match);
+    return isNaN(mediaId) ? null : {mediaId};
+  }
+
+  @memoize()
+  protected static loadAPI(): any{
+    return  loadScript('VIMEO_API_SOURCE', 'https://player.vimeo.com/api/player.js');
+  }
+
+  protected static mapOptions(cfg: MediaProviderConfig, component: ESLMedia): Options {
+    const options: Options ={};
+    if(cfg.mediaId) options.id = +cfg.mediaId;
+    Object.assign(options, cfg);
+    return options;
+  }
+
+  protected static buildIframe(cfg: MediaProviderConfig) {
+    const el = document.createElement('iframe');
+    el.id = 'esl-media-vimeo-' + randUID();
+    el.className = 'esl-media-inner esl-media-vimeo';
+    el.title = cfg.title;
+    el.setAttribute('aria-label', el.title);
+    el.setAttribute('data-vimeo-id', cfg.mediaId!);
+    el.setAttribute('src', 'https://player.vimeo.com/video/' + cfg.mediaId!);
```

Comment:
The origin `https://player.vimeo.com` occurs several times in code. Maybe it makes sense to define it as constant?